src/board.rs

```diff
@@ -142,6 +142,136 @@ impl Chessboard {
         println!();
     }
 
+    /// Converts a square index (0-63) to its corresponding chess rank and file.
+    ///
+    /// # Arguments
+    ///
+    /// * `square` - The square index (0-63) on the chessboard.
+    ///
+    /// # Returns
+    ///
+    /// A tuple containing the corresponding file (character) and rank (usize) for the given square.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// let (file, rank) = square_to_rank_file(35);
+    /// println!("File: {}, Rank: {}", file, rank);
+    /// // Output: File: 'D', Rank: 5
+    /// ```
+    pub fn square_to_rank_file(square: u8) -> (char, usize) {
+        let row = (square - 1) / 8 + 1;
+        let col = (square - 1) % 8;
+        let file = (b'A' + col) as char;
+        (file, row as usize)
+    }
+    /// Converts a chess rank and file to its corresponding square index (0-63).
```

Comment:
Add whitespace here. Can't believe the formatter okay'd this tbh, it's the worst thing I've ever seen in my life.